src/geventhttpclient/connectionpool.py

```diff
@@ -204,8 +204,13 @@ def _create_tcp_socket(self, family, socktype, protocol):
                 family, socktype, protocol)
 
             if self.ssl_context_factory is None:
-                ssl_options = self.default_options.copy()
-                ssl_options.update(self.ssl_options)
+                ssl_options = {}
+                if not self.insecure:
+                    ssl_options = self.default_options.copy()
+                    ssl_options.update(self.ssl_options)
```

Comment:
I'm not sure if we keep ssl_options data and pass it to gevent.ssl.wrap_socket will work with insecure mode. It could also run into ssl mode and check the key or something.